### PR TITLE
Parser + Tokenizer

### PR DESCRIPTION
**Pull request by ENDERZOMBI102**
In this PR i've worked on the parser and the tokenizer to make them work and output somewhat decent error messages

### CODE REVIEW

File: src/main/kotlin/dev/wisplang/wisp/ast/Type.kt [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
-interface BasicType
+interface BaseType {
+    companion object {
+        fun findType(name: String): BaseType {
+            if (name.isEmpty())
+                return VoidType.Void
 
-enum class PrimitiveTypes : BasicType {
+            for (type in PrimitiveTypes.values())
+                if (type.name.lowercase() == name)
```
Comment (Oliver-makes-code) [inline src/main/kotlin/dev/wisplang/wisp/ast/Type.kt:10]: Primitives shouldn't be case insensitive

File: src/main/kotlin/dev/wisplang/wisp/ast/Type.kt [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+}
+
+enum class PrimitiveTypes : BaseType {
     U1,
-    U8, I8,
+    U8,  I8,
     U16, I16,
     U32, I32, F32,
-    U64, I64, F64
+    U64, I64, F64;
```
Comment (Oliver-makes-code) [inline src/main/kotlin/dev/wisplang/wisp/ast/Type.kt]: Formatting is a little weird

File: src/main/kotlin/dev/wisplang/wisp/tokenizer/Keywords.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    IF, IMP,
+    IN, IS,
+    MATCH, PRIM,
+    TYPE, VAR,
+    WHILE;
+
+    companion object {
+        private val STRINGS: List<String> = Keywords.values().map { it.name.lowercase() }
+
+        fun strings() = STRINGS
```
Comment (Oliver-makes-code) [inline src/main/kotlin/dev/wisplang/wisp/tokenizer/Keywords.kt]: This isn't kotlin convention, you could make STRINGS public and just access that

**Review by Oliver-makes-code — CHANGES_REQUESTED** [review]
A couple changes, otherwise lgtm

File: src/main/kotlin/dev/wisplang/wisp/ast/Type.kt (hunk 1 above)
Comment (ENDERZOMBI102) [reply src/main/kotlin/dev/wisplang/wisp/ast/Type.kt:10]: they... are not case insensitive?
read the code, i'm lowercasing the enum names, not the parameter

File: src/main/kotlin/dev/wisplang/wisp/ast/Type.kt (hunk 2 above)
Comment (ENDERZOMBI102) [reply src/main/kotlin/dev/wisplang/wisp/ast/Type.kt]: reformatted

File: src/main/kotlin/dev/wisplang/wisp/tokenizer/Keywords.kt (hunk 3 above)
Comment (ENDERZOMBI102) [reply src/main/kotlin/dev/wisplang/wisp/tokenizer/Keywords.kt]: k

**Review by Oliver-makes-code — APPROVED** [review]
Looks good

**Review by violet-sh — APPROVED** [review]
LGTM

**Comment by ENDERZOMBI102** [comment]
welp me mrg

**Pull request merged**